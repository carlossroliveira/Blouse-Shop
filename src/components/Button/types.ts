// -------------------------------------------------
// Packages
// -------------------------------------------------
import { ButtonHTMLAttributes } from 'react';
// -------------------------------------------------
// Types
// -------------------------------------------------

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
}
