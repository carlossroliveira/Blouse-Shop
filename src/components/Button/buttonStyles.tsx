// -------------------------------------------------
// Packages
// -------------------------------------------------
import styled from 'styled-components';

export const ContainerSC = styled.section`
  display: flex;
  justify-content: center;
`;

export const ButtonSC = styled.button`
  all: unset;
  width: 100px;
  height: 30px;
  font-size: 16px;
  background: transparent;
  border: none;
  position: relative;
  color: ${({ theme }) => theme.color.secondary};
  cursor: pointer;
  z-index: 1;
  padding: 10px 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  font-weight: 600;

  &::after,
  &::before {
    content: '';
    position: absolute;
    bottom: 0;
    right: 0;
    z-index: -99999;
    transition: all 0.4s;
  }

  &::before {
    transform: translate(0%, 0%);
    width: 100%;
    height: 100%;
    background-color: ${({ theme }) => theme.color.primary};

    border-radius: 10px;
  }

  &::after {
    transform: translate(10px, 10px);
    width: 35px;
    height: 35px;
    background-color: ${({ theme }) => theme.color.tertiary};
    backdrop-filter: blur(5px);
    -webkit-backdrop-filter: blur(5px);
    border-radius: 50px;
  }

  &:hover::before {
    transform: translate(5%, 20%);
    width: 110%;
    height: 110%;
  }

  &:hover::after {
    border-radius: 10px;
    transform: translate(0, 0);
    width: 100%;
    height: 100%;
  }

  &:active::after {
    transition: 0s;
    transform: translate(0, 5%);
  }
`;
