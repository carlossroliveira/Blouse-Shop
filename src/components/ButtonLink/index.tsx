// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, LinkSC } from './buttonLinkStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonLinkProps } from './types';

export const ButtonLink = (props: ButtonLinkProps) => {
  return (
    <ContainerSC>
      <LinkSC to={`${props.path}`} onClick={props.onClick}>
        {props.text}
      </LinkSC>
    </ContainerSC>
  );
};
