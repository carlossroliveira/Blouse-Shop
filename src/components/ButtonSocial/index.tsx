// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonSocialStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonSocialProps } from './types';

export const ButtonSocial = (props: ButtonSocialProps) => {
  return (
    <ButtonSC role="button">
      <>
        {props.icon}
        {props.text}
      </>
    </ButtonSC>
  );
};
