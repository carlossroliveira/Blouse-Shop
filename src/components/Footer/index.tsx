// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ParagraphSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IFooterProps } from './types';

export const Footer = (props: IFooterProps) => {
  const currentYear: number = new Date().getFullYear();

  return (
    <ContainerSC>
      <ParagraphSC Color={props.color}>
        &copy; {currentYear} | Carlos Oliveira
      </ParagraphSC>
    </ContainerSC>
  );
};
