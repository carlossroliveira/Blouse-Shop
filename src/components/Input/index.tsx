// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './inputStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { InputProps } from './types';

export const Input = (props: InputProps) => {
  return (
    <ContainerSC>
      <input
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
    </ContainerSC>
  );
};
