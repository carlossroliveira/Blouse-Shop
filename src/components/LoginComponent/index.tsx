import React, { FormEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';

export const LoginComponent = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const NAVIGATION = useNavigate();
  const auth = useAuth();

  const handleSubmita = async (event: FormEvent) => {
    event.preventDefault();

    console.log(email);
    console.log(password);
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  };

  /*  async function onFinish(event: any, email: string, password: string) {
    event.preventDefault();
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  } */

  return (
    <>
      <form onSubmit={handleSubmita}>
        <div>
          <input
            type="text"
            placeholder="email"
            value={email}
            onChange={(event) => setEmail(event.target.value)}
          />
        </div>
        <div>
          <input
            type="text"
            placeholder="Senha"
            value={password}
            onChange={(event) => setPassword(event.target.value)}
          />
        </div>
        <button type="submit">Logar</button>
      </form>
    </>
  );
};
