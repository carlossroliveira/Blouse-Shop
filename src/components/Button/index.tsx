// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ContainerSC>
      <ButtonSC role="button">{props.text}</ButtonSC>
    </ContainerSC>
  );
};
