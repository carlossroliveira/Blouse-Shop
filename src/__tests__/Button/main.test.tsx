// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../components/Button';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(Button);

describe('Testing error button', () => {
  it('Should render the button correctly', () => {
    render(<LoginWithTheme text={'Button'} />);

    expect(
      screen.getByRole('button', {
        name: /button/i,
      }),
    ).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme text={'Button'} />);
    expect(render(<LoginWithTheme text={'Button'} />)).toMatchSnapshot();
  });
});
