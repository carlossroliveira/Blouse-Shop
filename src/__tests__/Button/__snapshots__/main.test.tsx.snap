// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error button component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  all: unset;
  width: 100px;
  height: 30px;
  font-size: 16px;
  background: transparent;
  border: none;
  position: relative;
  color: #383838;
  cursor: pointer;
  z-index: 1;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  font-weight: 600;
}

.c1::after,
.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
  z-index: -99999;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.c1::before {
  -webkit-transform: translate(0%,0%);
  -ms-transform: translate(0%,0%);
  transform: translate(0%,0%);
  width: 100%;
  height: 100%;
  background-color: #f8f8f8;
  border-radius: 10px;
}

.c1::after {
  -webkit-transform: translate(10px,10px);
  -ms-transform: translate(10px,10px);
  transform: translate(10px,10px);
  width: 35px;
  height: 35px;
  background-color: #90caf9;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-backdrop-filter: blur(5px);
  border-radius: 50px;
}

.c1:hover::before {
  -webkit-transform: translate(5%,20%);
  -ms-transform: translate(5%,20%);
  transform: translate(5%,20%);
  width: 110%;
  height: 110%;
}

.c1:hover::after {
  border-radius: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  width: 100%;
  height: 100%;
}

.c1:active::after {
  -webkit-transition: 0s;
  transition: 0s;
  -webkit-transform: translate(0,5%);
  -ms-transform: translate(0,5%);
  transform: translate(0,5%);
}

<body>
    <div>
      <section
        class="c0"
      >
        <button
          class="c1"
          role="button"
        >
          Button
        </button>
      </section>
    </div>
    <div>
      <section
        class="c0"
      >
        <button
          class="c1"
          role="button"
        >
          Button
        </button>
      </section>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  all: unset;
  width: 100px;
  height: 30px;
  font-size: 16px;
  background: transparent;
  border: none;
  position: relative;
  color: #383838;
  cursor: pointer;
  z-index: 1;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  font-weight: 600;
}

.c1::after,
.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
  z-index: -99999;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.c1::before {
  -webkit-transform: translate(0%,0%);
  -ms-transform: translate(0%,0%);
  transform: translate(0%,0%);
  width: 100%;
  height: 100%;
  background-color: #f8f8f8;
  border-radius: 10px;
}

.c1::after {
  -webkit-transform: translate(10px,10px);
  -ms-transform: translate(10px,10px);
  transform: translate(10px,10px);
  width: 35px;
  height: 35px;
  background-color: #90caf9;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-backdrop-filter: blur(5px);
  border-radius: 50px;
}

.c1:hover::before {
  -webkit-transform: translate(5%,20%);
  -ms-transform: translate(5%,20%);
  transform: translate(5%,20%);
  width: 110%;
  height: 110%;
}

.c1:hover::after {
  border-radius: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  width: 100%;
  height: 100%;
}

.c1:active::after {
  -webkit-transition: 0s;
  transition: 0s;
  -webkit-transform: translate(0,5%);
  -ms-transform: translate(0,5%);
  transform: translate(0,5%);
}

<div>
    <section
      class="c0"
    >
      <button
        class="c1"
        role="button"
      >
        Button
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing error buyBlouse component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  all: unset;
  width: 100px;
  height: 30px;
  font-size: 16px;
  background: transparent;
  border: none;
  position: relative;
  color: #383838;
  cursor: pointer;
  z-index: 1;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  font-weight: 600;
}

.c1::after,
.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
  z-index: -99999;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.c1::before {
  -webkit-transform: translate(0%,0%);
  -ms-transform: translate(0%,0%);
  transform: translate(0%,0%);
  width: 100%;
  height: 100%;
  background-color: #f8f8f8;
  border-radius: 10px;
}

.c1::after {
  -webkit-transform: translate(10px,10px);
  -ms-transform: translate(10px,10px);
  transform: translate(10px,10px);
  width: 35px;
  height: 35px;
  background-color: #90caf9;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-backdrop-filter: blur(5px);
  border-radius: 50px;
}

.c1:hover::before {
  -webkit-transform: translate(5%,20%);
  -ms-transform: translate(5%,20%);
  transform: translate(5%,20%);
  width: 110%;
  height: 110%;
}

.c1:hover::after {
  border-radius: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  width: 100%;
  height: 100%;
}

.c1:active::after {
  -webkit-transition: 0s;
  transition: 0s;
  -webkit-transform: translate(0,5%);
  -ms-transform: translate(0,5%);
  transform: translate(0,5%);
}

<body>
    <div>
      <section
        class="c0"
      >
        <button
          class="c1"
          role="button"
        >
          Button
        </button>
      </section>
    </div>
    <div>
      <section
        class="c0"
      >
        <button
          class="c1"
          role="button"
        >
          Button
        </button>
      </section>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  all: unset;
  width: 100px;
  height: 30px;
  font-size: 16px;
  background: transparent;
  border: none;
  position: relative;
  color: #383838;
  cursor: pointer;
  z-index: 1;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  font-weight: 600;
}

.c1::after,
.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
  z-index: -99999;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.c1::before {
  -webkit-transform: translate(0%,0%);
  -ms-transform: translate(0%,0%);
  transform: translate(0%,0%);
  width: 100%;
  height: 100%;
  background-color: #f8f8f8;
  border-radius: 10px;
}

.c1::after {
  -webkit-transform: translate(10px,10px);
  -ms-transform: translate(10px,10px);
  transform: translate(10px,10px);
  width: 35px;
  height: 35px;
  background-color: #90caf9;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-backdrop-filter: blur(5px);
  border-radius: 50px;
}

.c1:hover::before {
  -webkit-transform: translate(5%,20%);
  -ms-transform: translate(5%,20%);
  transform: translate(5%,20%);
  width: 110%;
  height: 110%;
}

.c1:hover::after {
  border-radius: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  width: 100%;
  height: 100%;
}

.c1:active::after {
  -webkit-transition: 0s;
  transition: 0s;
  -webkit-transform: translate(0,5%);
  -ms-transform: translate(0,5%);
  transform: translate(0,5%);
}

<div>
    <section
      class="c0"
    >
      <button
        class="c1"
        role="button"
      >
        Button
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
